nemo_skills/training/openrlhf/llm_reward.py

```diff
@@ -1,21 +1,40 @@
 import torch
-from nemo_skills.inference.server.model import get_model
-from nemo_skills.prompt.utils import get_prompt
+
 from nemo_skills.code_execution.math_grader import extract_answer
 from nemo_skills.evaluation.metrics.utils import is_correct_judgement
+from nemo_skills.inference.server.model import get_model
+from nemo_skills.prompt.utils import get_prompt
+from nemo_skills.utils import prefill_judgement
 
 
 def reward_func(queries: list[str], prompts: list[str], prompt_metadata: list[dict]):
-    expected_answers = [data["expected_answer"] for data in prompt_metadata]
-    predicted_answers = [extract_answer(query) for query in queries]
-    problems = [data["problem"] for data in prompt_metadata]
+    data_points = []
+    prefilled_judgements = []
+    prefilled_indices = set()
+    for metadata, query in zip(prompt_metadata, queries):
+        data_points.append(
+            {
+                "problem": metadata["problem"],
+                "expected_answer": metadata["expected_answer"],
+                "predicted_answer": extract_answer(query),
+            }
+        )
+        judgement = prefill_judgement(data_points[-1])
+        if judgement is not None:
+            prefilled_judgements.append(judgement)
+            prefilled_indices.add(len(data_points) - 1)
+
     llm = get_model(server_type="trtllm")
     prompt = get_prompt('judge/math', 'qwen-instruct')
```

Comment:
`server_type`, `prompt_config`, and `prompt_template` should not be hardcoded in the function. Make them parameters ideally.

Reply:
in a follow-up pr